agent/testagent.go

```diff
@@ -122,32 +115,45 @@ func NewTestAgentWithFields(t *testing.T, start bool, ta TestAgent) *TestAgent {
 
 // Start starts a test agent. It returns an error if the agent could not be started.
 // If no error is returned, the caller must call Shutdown() when finished.
-func (a *TestAgent) Start() (err error) {
+func (a *TestAgent) Start(t *testing.T) (err error) {
 	if a.Agent != nil {
 		return fmt.Errorf("TestAgent already started")
 	}
 
+	name := a.Name
+	// Many tests set Name to t.Name(), which takes up a lot of space at the
+	// start of the log messages. The only time we need to care about a name is
+	// when a test using multiple TestAgents.
+	// As a temporary workaround we ignore the default name and use a shorter
+	// default value. Tests which set a custom name for multiple agents will
+	// use the provided name.
+	// TODO: remove TestAgent.Name and accept a name arg on Start, to remove
+	// this workaround.
```

Comment:
I can make this change after the PR merges. It should be easy to automate, but will change ~800 lines. I did not want all of those changes distracting from the logging changes in this PR.